src/lib/rose_tree/rose_tree.ml

```diff
@@ -1,6 +1,6 @@
 open Core_kernel
 
-type 'a t = T of 'a * 'a t list
+type 'a t = T of 'a * 'a t list [@@deriving yojson]
```

Comment:
Because of the way that yojson derives ADTs, you probably want to introduce a custom yojson interface here.

Reply:
the deriving doesn't look so awful. you would omit the `T` constructor name? and probably turn this into an `{value:X,children:[]}`?

Reply:
Yeah, basically, but if you are fine with the deriving version, we can keep it. Mostly preference from the perspective of reading and writing filters.